hub/api/dataset.py

```diff
@@ -194,10 +202,89 @@ def like(
+
+        Raises:
+            InvalidPathException: If the source directory does not exist.
+            SamePathException: If the source and destination path are same.
+        """
+        if not os.path.isdir(src):
+            raise InvalidPathException(src)
+
+        if os.path.isdir(src) and os.path.isdir(dest):
+            if os.path.samefile(src, dest):
+                raise SamePathException(src)
+
+        ds = hub.dataset(dest)
+
+        unstructured = ImageClassification(source=src)
+
+        # TODO auto detect file extension
+        unstructured.structure(
+            ds, image_tensor_args={"sample_compression": compression}
+        )
+
+        return ds
+
+    @staticmethod
+    def ingest_kaggle(
+        tag: str,
+        src: str,
+        dest: str,
+        src_creds: dict,
+        compression: str,
+        overwrite: bool = False,
+    ) -> Dataset:
+        """Download and ingest a kaggle dataset and store it as a structured dataset to destination
+
+        Note:
+            Currently only local source path and only image datasets are supported.
+
+        Args:
+            tag (str): Kaggle dataset tag. Example: `"coloradokb/dandelionimages"` points to https://www.kaggle.com/coloradokb/dandelionimages
+            src (str): Local path to where the unstructured dataset is stored.
+            dest (str): Destination path where the structured dataset will be stored. Can be:-
+                - a Hub cloud path of the form hub://username/datasetname. To write to Hub cloud datasets, ensure that you are logged in to Hub (use 'activeloop login' from command line)
+                - an s3 path of the form s3://bucketname/path/to/dataset. Credentials are required in either the environment or passed to the creds argument.
+                - a local file system path of the form ./path/to/dataset or ~/path/to/dataset or path/to/dataset.
+                - a memory path of the form mem://path/to/dataset which doesn't save the dataset but keeps it in memory instead. Should be used only for testing as it does not persist.
+            src_creds (dict): A dictionary containing credentials used to access the dataset at the path.
+            compression (str): Compression type of dataset.
+            overwrite (bool): WARNING: If set to True this overwrites the dataset if it already exists. This can NOT be undone! Defaults to False.
+
+        Raises:
+            SamePathException: If the source and destination path are same.
+        """
+        if os.path.isdir(src) and os.path.isdir(dest):
+            if os.path.samefile(src, dest):
+                raise SamePathException(src)
+
+        download_kaggle_dataset(tag, local_path=src)
+        ds = hub.dataset(dest)
+        unstructured = ImageClassification(source=src)
+
+        # TODO auto detect file extension
+        unstructured.structure(
+            ds, image_tensor_args={"sample_compression": compression}
+        )
+
```

Comment:
you should not re-implement code when it exists in another function. instead, just call `hub.ingest(...)` after downloading the kaggle dataset